packages/select/src/components/query-list/itemRenderer.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import { MouseEventHandler } from "react";
+
+export interface IItemModifiers {
+    /** Whether this is the "active" (focused) item, meaning keyboard interactions will act upon it. */
+    active: boolean;
+
+    /** Whether this item is disabled and should ignore interactions. */
+    disabled: boolean;
+
+    // TODO: not happy with this name
+    /** Whether this item matches the predicate. A typical renderer could hide `false` values. */
+    filtered: boolean;
+}
+
+/**
+ * An object describing how to render a particular item.
+ * An `itemRenderer` receives the item as its first argument, and this object as its second argument.
+ */
+export interface IItemRendererProps {
```

Comment:
What's the benefit of splitting the props/modifier? Why is it no longer generic?

Reply:
it was only generic for the `item: T` field -- everything else has well-defined types. 
also `item` is the only user-supplied field; the others are derived from other props and state. so now the derived fields are in a separate place and can be reused across renders, rather than being forced to create a new object just to change `item`.